plugin/src/test/java/io/jenkins/plugins/casc/vault/VaultTestUtil.java

```diff
@@ -0,0 +1,76 @@
+
+public class VaultTestUtil {
+    private final static Logger LOGGER = Logger.getLogger(VaultTestUtil.class.getName());
+
+    public static final String VAULT_DOCKER_IMAGE = "vault:1.0.3";
+    public static final String VAULT_ROOT_TOKEN = "root-token";
+    public static final String VAULT_USER = "admin";
+    public static final String VAULT_URL = "http://localhost:8200";
+    public static final String VAULT_PW = "admin";
+    public static final String VAULT_POLCY= "io/jenkins/plugins/casc/vault/vaultTest_adminPolicy.hcl";
+    public static final String VAULT_PATH_V1 = "kv-v1/admin";
+    public static final String VAULT_PATH_V2 = "kv-v2/admin";
+
+    private static Container.ExecResult runCommand(VaultContainer container, final String... command) throws IOException, InterruptedException {
+        LOGGER.log(Level.FINE, String.join(" ", command));
+        final Container.ExecResult result = container.execInContainer(command);
+        final String out = result.getStdout();
+        final String err = result.getStderr();
+        if (out != null && !out.isEmpty()) {
+            LOGGER.log(Level.FINE, result.getStdout());
+        }
+        if (err != null && !err.isEmpty()) {
+            LOGGER.log(Level.WARNING, result.getStderr());
+        }
+        return result;
+    }
+
+    public static boolean isWindowsNode() {
+        return System.getProperty( "os.name" ).startsWith( "Windows" );
+    }
+
+    public static VaultContainer createVaultContainer() {
+        if (isWindowsNode()) return null;
+        return new VaultContainer<>(VaultTestUtil.VAULT_DOCKER_IMAGE)
+                .withVaultToken(VaultTestUtil.VAULT_ROOT_TOKEN)
+                .withClasspathResourceMapping(VAULT_POLCY, "/admin.hcl", BindMode.READ_ONLY)
+                .withVaultPort(8200);
+    }
+
+    public static void configureVaultContainer(VaultContainer container) {
+        if (isWindowsNode()) return;
+        try {
+            // TODO: proper life cycle or polling
+            // Give vault 5s to start
+            Thread.sleep(5000);
+        } catch (InterruptedException e) {}
+        try {
+            // Create Secret Backends
+            runCommand(container, "vault", "secrets", "enable", "-path=kv-v2", "-version=2", "kv");
+            runCommand(container, "vault", "secrets", "enable", "-path=kv-v1", "-version=1", "kv");
+
+            // Create user/password credential
+            runCommand(container, "vault", "auth", "enable", "userpass");
+            runCommand(container, "vault", "write", "auth/userpass/users/" + VAULT_USER, "password=" + VAULT_PW, "policies=admin");
+
+            // Create policies
+            runCommand(container, "vault", "policy", "write", "admin", "/admin.hcl");
+
+            // add secrets for v1 and v2
+            runCommand(container, "vault", "kv", "put", VAULT_PATH_V1, "key1=123", "key2=456");
+            runCommand(container, "vault", "kv", "put", VAULT_PATH_V2, "key1=123", "key2=456");
+
```

Comment:
I would love to see an approle test too :smile:

Reply:
Yes I would like them too, but im a little stuck here. 

As I mentioned in the PR ToDos:
```
AppRole login tests
Approle id and secret are generated and queried after vault is running.
However, the env vars are already set and defined BEFORE vault is bootstrapped.
```

So the problem is that I get the AppRoleID/Secret only after all the env vars are set in `@Rule`. Is there a way to adjust the env vars in the `@Test` annotated function?

Reply:
I went straight for code review, skipped the text wall :smile:

Reply:
You could skip using the `envRule` we have examples of how to set envs and call configure afterwards in our tests, as I recall.

Reply:
AppRoles are added now :)